codegen/src/test/resources/software/amazon/awssdk/codegen/poet/model/alltypesrequest.java

```diff
@@ -495,63 +495,63 @@ private AllTypesRequest(BuilderImpl builder) {
 
     /**
      * Returns the value of the StringMember property for this object.
-     *
+     * 
      * @return The value of the StringMember property for this object.
      */
-    public String stringMember() {
+    public final String stringMember() {
         return stringMember;
     }
 
     /**
      * Returns the value of the IntegerMember property for this object.
-     *
+     * 
      * @return The value of the IntegerMember property for this object.
      */
-    public Integer integerMember() {
+    public final Integer integerMember() {
         return integerMember;
     }
 
     /**
      * Returns the value of the BooleanMember property for this object.
-     *
+     * 
      * @return The value of the BooleanMember property for this object.
      */
-    public Boolean booleanMember() {
+    public final Boolean booleanMember() {
         return booleanMember;
     }
 
     /**
      * Returns the value of the FloatMember property for this object.
-     *
+     * 
      * @return The value of the FloatMember property for this object.
      */
-    public Float floatMember() {
+    public final Float floatMember() {
         return floatMember;
     }
 
     /**
      * Returns the value of the DoubleMember property for this object.
-     *
+     * 
      * @return The value of the DoubleMember property for this object.
      */
-    public Double doubleMember() {
+    public final Double doubleMember() {
```

Comment:
It'd nice if we can only add `final` to non-final model classes, but I guess it's not trivial to do so with the current implementation of the model codegen class spec.

Reply:
Yeah special casing events would make `AwsServieModel` even more complicated. Since they're `final` classes we can always remove the `final` on the methods at a later time without breaking binary compatibility.